shotover/src/unix_socket_server.rs

```diff
@@ -0,0 +1,128 @@
+use crate::hot_reload::{Request, Response};
+use serde_json;
+use std::collections::HashMap;
+use std::path::Path;
+use tokio::io::{AsyncBufReadExt, AsyncWriteExt, BufReader};
+use tokio::net::{UnixListener, UnixStream};
+
+pub struct UnixSocketServer {
+    socket_path: String,
+    listener: Option<UnixListener>,
+}
+
+impl UnixSocketServer {
+    pub fn new(socket_path: String) -> Self {
+        Self {
+            socket_path,
+            listener: None,
+        }
+    }
+
+    pub async fn start(&mut self) {
+        if Path::new(&self.socket_path).exists() {
+            std::fs::remove_file(&self.socket_path).unwrap();
+        }
+        let listener = UnixListener::bind(&self.socket_path).unwrap();
+        self.listener = Some(listener);
+    }
+
+    pub async fn run(&mut self) {
+        let listener = self.listener.as_ref().unwrap();
+        loop {
+            let (stream, _) = listener.accept().await.unwrap();
+            self.handle_connection(stream).await;
+        }
+    }
+
+    async fn handle_connection(&self, stream: UnixStream) {
```

Comment:
For error handling here, I recommend making this function return an `anyhow::Result`.
Then the function body should return errors using the `?` operator: See: [TRPL](https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator), [Rust reference](https://doc.rust-lang.org/reference/expressions/operator-expr.html#the-question-mark-operator)
And then in `run` add some logic to log the error and continue the loop.